release_creation/snapshot/create.py

```diff
@@ -0,0 +1,64 @@
+from typing import Dict, List, Optional, Tuple
+from semantic_version import Version
+import os
+import platform
+import subprocess
+import shutil
+
+
+_OUTPUT_ARCHIVE_FILE_BASE = "dbt-core-all-adapters-snapshot"
+_FILE_DIR = os.path.dirname(os.path.realpath(__file__))
+_SNAPSHOT_FILE = f"{_FILE_DIR}/snapshot.requirements.txt"
+
+
+def _get_local_platform() -> str:
+    local_sys = platform.system()
+    if local_sys == "Linux":
+        return "linux"
+    elif local_sys == "Windows":
+        return "windows"
+    elif local_sys == "Darwin":
+         return "mac"
+    else:
+        raise ValueError(f"Unsupported system {local_sys}")
+
+
+def _get_requirements_prefix(
+    major_version: Optional[int], minor_version: Optional[int], is_pre: bool = False
+):
+    suffix = "latest"
+    if is_pre:
+        suffix = "pre"
+    return f"v{major_version}.{minor_version}.{suffix}"
+
+
+def _generate_download_command_args(
+    requirements_prefix: str, is_pre: bool = False
+) -> str:
+    download_command = ""
+    if is_pre:
```

Comment:
Nit: My gut tells me its cleaner to have a list that gets joined rather than building the string on top of itself. Less chance for malforming the variable?

Reply:
I'm 50/50 here but you've cast the deciding vote!